pkg/plugin/v3/init.go

```diff
@@ -92,6 +102,11 @@ func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
 		"defaults to the go package of the current working directory.")
 	fs.StringVar(&p.config.Domain, "domain", "my.domain", "domain for groups")
 	fs.StringVar(&p.config.ProjectName, "project-name", "", "name of this project")
+
+	fs.StringVar(&p.crdVersion, "crd-version", defaultCRDVersion,
+		"CustomResourceDefinition version: [v1, v1beta1]")
+	fs.StringVar(&p.webhookVersion, "webhook-version", defaultWebhookVersion,
+		"{Mutating,Validating}WebhookConfiguration version: [v1, v1beta1]")
```

Comment:
what about use in both cases api-version?

Reply:
What if I want v1beta1 CRDs but v1 webhooks?

Reply:
would it really make sense? Would it work?

Reply:
IHMO we should NOT add these flags.
The init ONLY scaffold the project.
The API version is part of the creates API domain.

Reply:
That isn't strictly true since we scaffold CRD and webhook patch files in `init`. More importantly though, the user experience is worse/ambiguous with `create api --crd-version`/`create webhook --webhook-version` because
1. They have to remember to type it every time if they want a non-default version
1. It implies that they can have multiple CRD/webhook API versions per project, which isn't allowed

Setting these values once in `init --crd-version --webhook-version` avoids both potential issues.

Reply:
My suggestion is : 
- create api --api-version
- create webhook --api-version
where users only would use `api-version` if do not want the default value. 

If I as a dev would like to consume the `create api` and `init` plugin but not the webhooks, how I can do that when the init plugin has flags for webhooks? If we would like to define that the webhooks by default would use version X and CRDs Y, how we can do that if it is in the init?

Reply:
If you're saying that a project author may choose to not want webhooks on `init` but want to implement them later, then yes that may be a problem under my proposed scheme. However if they are making that choice later, then likely they will be ok with using the default version _unless_ they do have some use case for using a non-default version; they'd have to make manual changes under my proposed flag scheme in the latter case.

Your proposed scheme doesn't fix the problem of `init` scaffolding a webhook patch, for which we need the webhook version. Perhaps we can consider scaffolding that file on `create webhook`. I'm still not a fan of having to specify `--webhook-version` each time you want a non-default webhook version.

Reply:
> Your proposed scheme doesn't fix the problem of init scaffolding a webhook patch, for which we need the webhook version. Perhaps we can consider scaffolding that file on create webhook

I am completed agree with you that `enablecainection_patch.go` ought to be scaffolded in the `create webhook` since kb plugin design allows us to consume 1..N plugins and to respect the responsibility domains. 
I will track an issue for we change that. 

> I'm still not a fan of having to specify --webhook-version each time you want a non-default webhook version.
I think has a misunderstanding. The suggestion is: 


```go
const defaultApiVersion = "v1"
func (p *createWebhookPlugin) BindFlags(fs *pflag.FlagSet) {
	...
	fs.StringVar(&p.webhooks.ApiVersion, "api-version", defaultApiVersion, fmt.Sprintf("if not set, the default value is v1. Options [v1, v1beta1]", defaultApiVersion))
       ...
}
```

See that we have PR to store the webhooks already which is on hold until we reach a consensus: https://github.com/kubernetes-sigs/kubebuilder/pull/1696


Then, it means that **nothing will change for the users**.  I mean, they will still be using ONLY `kb create webhook`  or `create api` unless they need to scaffold the files with the old version. See:

### Default workflow 
 **nothing will change for the users and we will be encouraging the usage of the latest k8s version as well**
- kb create webhook 
- kb create api

### When users still requiring scaffold old apis
- kb create webhook --api-version
- kb create api --api-version

**NOTE**: IMO we should NOT use the nomenclatures `--webhook-version` and `crd-version` it is NOT aligned with the k8s api nomenclature which brings confusion.

Reply:
I think it is solved. So, I am closing this one.